### PR TITLE
TWAP index

### PR DESCRIPTION
**Pull request by vigan-abd**

### CODE REVIEW

File: src/twap.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+ * @param {(a) => string|number} selector Optional selector lambda, if not provided item will be used as value
+ * @returns {{ close: string, high: string, low: string, open: string }}
+ */
+const kline = (values, selector = null) => {
+  const last = values.length - 1
+
+  const close = nBN(selector ? selector(values[0]) : values[0]).toString()
+  const open = nBN(selector ? selector(values[last]) : values[last]).toString()
+  const low = min(values, selector)
+  const high = max(values, selector)
```
Comment (robertkowalski) [inline src/twap.js:60]: i would suggest being more strict on the data format.

there is no reason why high/low should not have a set index in the array, like open close... right?

Comment (robertkowalski) [reply src/twap.js:60]: also: simplifies the code...

and less fuzzy / ambigious / guessing logic

File: examples/twap.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+const { typicalPrice, kline, TWAP } = require('..')
+
+const primitiveData = ['10002', '10001', '10000']
+const complexData = [
+  { price: '10003', ts: 1578499254526 },
+  { price: '10001', ts: 1578499255526 },
+  { price: '10000', ts: 1578499256526 }
+]
+
+console.log(kline(primitiveData)) // { close: '10002', high: '10002', low: '10000', open: '10000' }
```
Comment (robertkowalski) [inline examples/twap.js:12]: hm, very strange. why does it extrapolate values out of incomplete data? we must have all ticker values anyway, there should be hopefully no need for this?

File: src/twap.js (hunk 1 above)
Comment (vigan-abd) [reply src/twap.js:60]: I agree, we can restrict the structure, just ts + price should be enough and readable

File: examples/twap.js (hunk 2 above)
Comment (robertkowalski) [reply examples/twap.js:12]: makes sense now. we use it on our grouped values by interval.

**Review by robertkowalski — APPROVED** [review]
lgtm

**Pull request merged**